### DISCUSSION TITLE
Midi Channel = Omni

### DISCUSSION THREAD

**Discussion by Wapata** [shanteacontrols/OpenDeck#148] [Ideas]
Hi !
I don't know how long your "to-do list" is, but if you mind, I still have some ideas. I hope it's okay. ;-)

We can choose MIDI channel from 1 - 16 but not "Omni".
Just two exemple (and we can find many more !) :

- Button MIDI CC 120 : Mutes all sounding notes => One emergency button to shout a sound wherever it come from.
- LED reacting to Midi Note On => a led that can be used whatever the channel used by a Midi device.

It's for standalone devices mostly. We can easily change the routing rules with a computer, but not with a simple keyboard or expander.

**Comment by paradajz** [comment]
I was actually planning on adding more settings for MIDI channels in the next release. However, the next release won't come for a while since I want to have Bluetooth support as well in it.

For the channels, I had the idea of following settings:

* Use global MIDI channel for all components
* Use per-block MIDI channel (e.g. use MIDI channel 1 for all buttons and channel 2 for all LEDs)
* Use individual channel settings for all components (the current system)

These settings are mainly convenience, though - just easier/quicker way to configure channels than the current implementation.

Omni channel for LEDs sounds reasonable. I don't understand your first bullet point, though.

**Comment by Wapata** [comment]
It look good, it look as I looked for :-)

I have an old midi expander (with old Yamaha general midi sounds) witch 16 channels. I can link it with 2 midi keyboard, 1 midi recorder diy and other diy controllers (as OpenDeck by instance). When "something" goes wrong, it make a ghost sound but... Nobody know witch controller did it.
And there is a midi cc to solve that. So I wanted a midi "stop" button that affect all channels. Each expander linked will kill all the "note on".

**Comment by paradajz** [comment]
In case you're willing to experiment... I've attached development version for Discovery board implementing this.

First of all, you should configure buttons/LEDs/whatever you want to MIDI channel 17. It's explained in the interface now what channel 17 does (buttons/analog/touchscreen/encoders: send specified message on all channels, LEDs: ignore specified channel/listen to everything).

If you want to use the button as an "kill switch" then that button needs to send Note Off only or CC with value 0 only. This wasn't possible until now, but I've added these new options under "message type" section for buttons, so select the one you want (you want just CC 0 from what I've gathered).

[discovery.zip](https://github.com/shanteacontrols/OpenDeck/files/8224444/discovery.zip)

Very lightly tested for now.

**Comment by paradajz** [comment]
Released: https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.0.0

**Reply by Wapata** [reply]
Damn, I didn't answer at your first proposal. Sorry, I'm a lot out of home because of work-travels.
I will put my hands on this as soon as I can. Thanks a lot for your work !
